low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -15,35 +15,116 @@
 # For any questions about this software or licensing,
 # please email opensource@seagate.com or cortx-questions@seagate.com
 
-"""
- ***************************************************************************
-  Description: ServerMap class provides resource map and related information
-               like health, manifest, etc,.
- ***************************************************************************
-"""
+
+import time
+import re
+import errno
+from framework.utils.ipmi_client import IpmiFactory
+from framework.base import sspl_constants
 
 from resource_map import ResourceMap
 
 
 class ServerMap(ResourceMap):
-    """Provides server resource related information."""
+    """ServerMap class provides resource map and related information
+    like health, manifest, etc,.
+    """
 
     name = "server"
 
     def __init__(self):
         """Initialize server"""
-        pass
+        super().__init__()
+        self._ipmi = IpmiFactory().get_implementor("ipmitool")
+        self.platform_sensor_list = ['Temperature', 'Voltage', 'Current']
+        self.server_frus = {
+            'platform_sensors': self.build_platform_sensor_data,
+        }
 
     def get_health_info(self, rpath):
         """
         Get health information of fru in given resource id
 
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
-        pass
+        fru_found = False
+        info = dict()
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+        if leaf_node == "compute":
+            for fru in self.server_frus:
+                info.update({fru: self.server_frus[fru]()})
+            info["last_updated"] = int(time.time())
+            fru_found = True
+        else:
+            fru = None
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_frus.get(fru):
+                    fru_found = True
+                    info = self.server_frus[fru]()
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                "Health provider doesn't have support for'{rpath}'.")
+        return info
+
+    @staticmethod
+    def get_node_details(node):
```

Comment:
Both these functions can be in resource_map class, refer 
https://github.com/Seagate/cortx-monitor/pull/419/files